monkey/infection_monkey/local_machine_info.py

```diff
@@ -0,0 +1,26 @@
+from pathlib import Path
+from typing import Optional
+
+from monkeytypes import MutableInfectionMonkeyBaseModel, OperatingSystem
+
+
+class LocalMachineInfo(MutableInfectionMonkeyBaseModel):
+    """
+    Contains information about the local machine
+
+    :param operating_system: Operating system of the local machine
+    :param temporary_directory: Path to a temporary directory on the local machine
+                                to store artifacts
+    :param interface_to_target: Interface on the local machine that can be reached
+                                by the target machine
+    """
+
+    operating_system: OperatingSystem
+    temporary_directory: Path
+    interface_to_target: Optional[str] = None
+
+    def set_interface_to_target(self, interface_to_target: str):
+        self.interface_to_target = interface_to_target
+
+    def reset_interface_to_target(self):
+        self.interface_to_target = None
```

Comment:
I don't think this is the right approach. The race condition in the puppet is a good example of why.

I think it would be better for this model to include an immutable collection of interfaces. It could also include `get_interface_to_target(self, target: ipv4address) -> ipv4interface)`.

Reply:
I'm not sure how the first suggestion would work. I went with the second one.

Reply:
Plugins may need to know what interfaces are on the system for an unforeseen reason. I think it would be good to include a collection of interfaces with the data structure. It can be populated when the object is created, the same way the OS and temporary directory are.

Reply:
Looks like we use `ifaddr` in `common.network_utils.get_network_interfaces()`. I think you can just use `get_network_interfaces()` in `monkey.py` to get the interfaces and pass them to the constructor.

Reply:
This would require us to modify Log4Shell to loop through the interfaces and find one that the target can connect to.

When would a plugin need all interfaces on the system? If we don't see a use case right now, I'd say we leave it as it is.